clients/drcachesim/tools/invariant_checker.cpp

```diff
@@ -454,7 +454,10 @@ invariant_checker_t::parallel_shard_memref(void *shard_data, const memref_t &mem
                   << " type " << memref.instr.type << "\n";
     }
     if (memref.marker.type == TRACE_TYPE_MARKER &&
-        memref.marker.marker_type == TRACE_MARKER_TYPE_TIMESTAMP) {
+        memref.marker.marker_type == TRACE_MARKER_TYPE_TIMESTAMP &&
+        // Timestamps at the start of a chunk should be invisible to analysis tools. These
+        // are duplicate timestamps used to faciliate skipping.
+        shard->prev_entry_.marker.marker_type != TRACE_MARKER_TYPE_CHUNK_FOOTER) {
```

Comment:
See comment above: I believe we can remove this as reader_t hides these.  Plus, given that reader_t hides the dup timestamps, it is possible that the first real entry in the chunk is a timestamp (I don't think the inserted dups are omitted in that case?), and then if you keep this check here you can have a false negative.  I think what we want to do is to allow a chunk footer in between a branch and its target.